### DISCUSSION TITLE
Proposta e valutazione per API mobile oriented.

### DISCUSSION THREAD

**Discussion by zerai** [iisgiua/giuaschool#154] [Idee e proposte]
**Genesi**: [dicussione](https://github.com/trinko/giuaschool/discussions/151)

**Proposta**: Sviluppo e documentazione (openApi) di Api per il mobile (Android/Ios)

**Stato attuale**:
Singolo controller `src/Controller/AppController` (verificare funzionalità test e doc.)

App esterna sviluppata da @HiemSword e @Franck1421 

**Potenzialmente interessati all'argomento** : 
- @MatteoGheza 
- @manuelbarberi  
- @HiemSword 
- @Franck1421 
- @zerai

**Collaboratori disponibili** (specifiche openApi, automazione test, sviluppo codice ecc)
- @zerai
- @MatteoGheza
- fatevi avanti!!!

Contribuire alla discussione sgnalando aspetti tecnici e feature da implementare lato api.

**Comment by zerai** [comment]
Considerare un sistema di autenticazione stateless del tipo JWT (no cookies)

Adesione agli standard odierni per quanto possibile.
- [REST](https://it.wikipedia.org/wiki/Representational_state_transfer)
- [OpenApi](https://swagger.io/docs/specification/about/) per doc. e specifiche
- [Api Problem](https://datatracker.ietf.org/doc/html/rfc7807) (gestione/mappatura errori)

**Comment by MatteoGheza** [comment]
@zerai grazie per aver aperto questa discussione da parte della community.
Penso anche io che sia utile aderire ad un qualche standard per l'autenticazione.
Aggiungimi pure all'elenco dei "Collaboratori disponibili"; sono un po' arrugginito sul lato "Symfony", ma potrei aiutare sul lato OpenAPI/JWT, o sul lato dell'"infrastruttura CI/CD", se si renderà necessario aggiornare anche quella.

**Comment by HiemSword** [comment]
Io, @Franck1421 e @Krek3r (gli attuali sviluppatori dell'app) pensiamo di rimanere concentrati a sviluppare l'app per Android (iOS per noi è impossibile a causa del costo per pubblicare l'app) ma ogni tanto potremmo dare una mano con l'API.

**Comment by zerai** [comment]
@trinko sarebbe possibile provare a migliorare la CD/CI?
mi riferisco ai lunghi tempi d'attesa per un feedback, mi pare che anche @MatteoGheza  abbia dato la sua disponibilià/eperienza per un'eventuale intervento se necessario.
Apro discussione separata da questa visto il tema?

**Reply by trinko** [reply]
> @trinko sarebbe possibile provare a migliorare la CD/CI? 

E' senz'altro meglio aprire una nuova discussione separata, in modo che chi è interessato possa partecipare.
Per quanto mi riguarda, ogni miglioramento è sempre ben accetto. Però è necessario fare una _pull request_ , in modo che io possa valutare l'eventuale impatto sul codice attualmente in uso.
